### DISCUSSION TITLE
How to tell pudb to pause on exception

### DISCUSSION THREAD

**Discussion by cool-RR** [inducer/pudb#488] [Q&A] [answered]
Hi everyone!

This might be a silly question from a long-time PuDB user, but for some reason I can't figure it out. How do I get PuDB to pause and show me the PuDB UI when there's an exception? Right now I'm using `pudb.set_trace()`, which pauses without an exception, which I don't want. If I use `pudb.set_trace(False)`, and an exception happens, it doesn't pause and start the PuDB UI, it just exits Python.

Thanks for your help,
Ram Rachum.

**Comment by inducer** [comment, marked as answer]
Two ways:
- Run your script with `python -m pudb myscript.py`.
- Catch the exception and run `pudb.pm()` in the `except` clause. (See #486 for some possible fixes to that.)

**Comment by cool-RR** [comment]
Thank you for the quick reply Andreas. Unfortunately I work in a corporate
environment where a complicated build process is running my Python program,
so I can't just slip `-m pudb` in there.

Regarding the second option, that would require knowing which exception it
is in advance :(

Is there no other way to do that?

**Reply by inducer** [reply]
>  Regarding the second option, that would require knowing which exception it is in advance :(

Can't you just catch `Exception`? (see the snippet in the description of #486 for an example)

**Comment by cool-RR** [comment]
Ah, I didn't consider wrapping the entire program! Yes, this sounds like it
could work. Thank you!
